contrib/pkg/broker/user_provided/controller/controller.go

```diff
@@ -104,8 +122,18 @@ func (c *userProvidedController) RemoveServiceInstance(id string) (*brokerapi.De
 	return &brokerapi.DeleteServiceInstanceResponse{}, nil
 }
 
-func (c *userProvidedController) Bind(instanceID string, bindingID string, req *brokerapi.BindingRequest) (*brokerapi.CreateServiceBindingResponse, error) {
-	cred := c.instanceMap[instanceID].Credential
+func (c *userProvidedController) Bind(
+	instanceID,
+	bindingID string,
+	req *brokerapi.BindingRequest,
+) (*brokerapi.CreateServiceBindingResponse, error) {
+	c.rwMutex.RLock()
+	defer c.rwMutex.RUnlock()
```

Comment:
I think you can just remove the "defer" and then move this line after the map look-up. Then we don't need to keep the lock any longer than necessary.  Granted the rest of the func is trivial but its a good habit to keep these things tight.